packages/strapi-plugin-graphql/services/utils.js

```diff
@@ -79,7 +79,7 @@ const amountLimiting = (params = {}) => {
 
   if (!amountLimit) return params;
 
-  if (!params.limit || params.limit === -1 || params.limit > amountLimit) {
+  if (_.isNil(params.limit) || params.limit === -1 || params.limit > amountLimit) {
```

Comment:
Why don't we simplify the statement `params.limit === -1` to `params.limit < 0` to get rid of the else if condition? 
I believe that we should have the same logic is we query with a limit of `-2` than with `-1`, what do you think?

Reply:
Actually  -1 is a specific value to ask for everything. other negative values are invalid

Reply:
Ok I see! thanks for your response